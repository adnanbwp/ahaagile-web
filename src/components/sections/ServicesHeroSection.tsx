import React from 'react';
import { Clock, DollarSign, Users, Zap } from 'lucide-react';
import { Card, CardContent } from '@/components/ui/card';
import { Badge } from '@/components/ui/badge';

interface ServicesHeroSectionProps {
  className?: string;
}

interface ProblemAreaProps {
  icon: React.ComponentType<{ className?: string }>;
  title: string;
  description: string;
}

const ProblemArea: React.FC<ProblemAreaProps> = ({ icon: Icon, title, description }) => (
  <Card className="card-elevated hover:card-brand group transition-all duration-300">
    <CardContent className="p-6 text-center">
      <div className="w-12 h-12 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center group-hover:bg-primary transition-colors duration-300">
        <Icon className="w-6 h-6 text-foreground group-hover:text-primary-foreground" />
      </div>
      <h3 className="text-lg font-heading font-semibold text-foreground group-hover:text-primary-foreground mb-2 transition-colors duration-300">
        {title}
      </h3>
      <p className="text-sm text-muted-foreground group-hover:text-primary-foreground/90 transition-colors duration-300">
        {description}
      </p>
    </CardContent>
  </Card>
);

export default function ServicesHeroSection({ className = '' }: ServicesHeroSectionProps): JSX.Element {
  const problemAreas: ProblemAreaProps[] = [
    {
      icon: Clock,
      title: "Time Drain",
      description: "Partners and senior staff spending 25+ hours weekly on email management and administrative tasks instead of billable client work."
    },
    {
      icon: DollarSign,
      title: "Billing Inefficiency", 
      description: "High-value professionals doing low-value work, reducing billable hour capacity and directly impacting revenue potential."
    },
    {
      icon: Users,
      title: "Client Frustration",
      description: "Delayed responses and communication gaps leading to client dissatisfaction and potential account losses."
    },
    {
      icon: Zap,
      title: "Work Interruption",
      description: "Constant context switching disrupting deep work and reducing overall productivity and quality of deliverables."
    }
  ];

  return (
    <div className={className}>
      {/* Hero Section */}
      <section className="relative py-20 lg:py-24 bg-[var(--gradient-primary)]">
        {/* Background Pattern Overlay */}
        <div className="absolute inset-0 opacity-10">
          <div className="absolute inset-0 bg-gradient-to-r from-transparent via-white to-transparent transform -skew-y-1"></div>
        </div>
        
        <div className="container-brand relative z-10">
          <div className="text-center max-w-4xl mx-auto">
            {/* Professional Services Badge */}
            <div className="mb-6">
              <Badge variant="secondary" className="bg-white/10 text-primary-foreground border-white/20 text-sm font-medium px-4 py-2">
                Professional Services Focus
              </Badge>
            </div>
            
            {/* Main Headline */}
            <h1 className="text-hero mb-6 text-primary-foreground">
              Intelligent Workflow Automation for Professional Services
            </h1>
            
            {/* Subheading */}
            <p className="text-subhero mb-8 text-foreground/80 max-w-3xl mx-auto">
              Reclaim 150+ Hours Per Employee Annually Through Intelligent Email Automation
            </p>
          </div>
        </div>
      </section>

      {/* Problem Statement Section */}
      <section className="section-spacing bg-gradient-to-br from-muted to-background">
        <div className="container-brand">
          <div className="text-center max-w-4xl mx-auto mb-16">
            <h2 className="text-3xl lg:text-4xl font-heading font-bold text-foreground mb-6">
              The Hidden Cost of Email-Driven Workflows
            </h2>
            
            {/* Financial Impact Callout */}
            <div className="bg-gradient-accent rounded-xl p-8 mb-8 text-center">
              <div className="text-4xl lg:text-5xl font-bold text-foreground mb-2">
                $236,000 to $330,000
              </div>
              <p className="text-lg font-medium text-foreground">
                Annual productivity loss for a typical 3-accountant professional services firm
              </p>
            </div>
            
            <p className="text-lg text-muted-foreground leading-relaxed">
              Email-driven workflows are silently draining your firm&apos;s profitability. While your team 
              juggles client communications, document management, and administrative tasks through 
              scattered email threads, valuable billable hours disappear into operational inefficiency.
            </p>
          </div>

          {/* Four Problem Areas Grid */}
          <div className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-4 gap-6">
            {problemAreas.map((area, index) => (
              <ProblemArea
                key={index}
                icon={area.icon}
                title={area.title}
                description={area.description}
              />
            ))}
          </div>
        </div>
      </section>
    </div>
  );
}