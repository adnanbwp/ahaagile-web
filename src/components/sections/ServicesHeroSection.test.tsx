import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ServicesHeroSection from './ServicesHeroSection';

describe('ServicesHeroSection', () => {
  beforeEach(() => {
    // Mock window.matchMedia for responsive tests
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  describe('Hero Section Rendering', () => {
    it('renders the main heading correctly', () => {
      render(<ServicesHeroSection />);
      
      expect(screen.getByRole('heading', { 
        name: /intelligent workflow automation for professional services/i 
      })).toBeInTheDocument();
    });

    it('renders the subheading with time savings', () => {
      render(<ServicesHeroSection />);
      
      expect(screen.getByText(/reclaim 150\+ hours per employee annually through intelligent email automation/i))
        .toBeInTheDocument();
    });

    it('displays professional services badge', () => {
      render(<ServicesHeroSection />);
      
      expect(screen.getByText(/professional services focus/i)).toBeInTheDocument();
    });

    it('applies gradient background styling', () => {
      render(<ServicesHeroSection />);
      
      const heroSection = screen.getByRole('heading', { 
        name: /intelligent workflow automation for professional services/i 
      }).closest('section');
      
      expect(heroSection).toHaveStyle({
        background: 'linear-gradient(135deg, var(--brand-navy) 0%, var(--brand-navy-light) 100%)'
      });
    });
  });

  describe('Problem Statement Section', () => {
    it('renders the problem statement header', () => {
      render(<ServicesHeroSection />);
      
      expect(screen.getByRole('heading', { 
        name: /the hidden cost of email-driven workflows/i 
      })).toBeInTheDocument();
    });

    it('displays financial impact callout with correct amounts', () => {
      render(<ServicesHeroSection />);
      
      expect(screen.getByText(/\$236,000 to \$330,000/)).toBeInTheDocument();
      expect(screen.getByText(/annual productivity loss for a typical 3-accountant professional services firm/i))
        .toBeInTheDocument();
    });

    it('shows problem description text', () => {
      render(<ServicesHeroSection />);
      
      expect(screen.getByText(/email-driven workflows are silently draining your firm's profitability/i))
        .toBeInTheDocument();
    });
  });

  describe('Four Problem Areas Grid', () => {
    it('renders all four problem area cards', () => {
      render(<ServicesHeroSection />);
      
      expect(screen.getByText('Time Drain')).toBeInTheDocument();
      expect(screen.getByText('Billing Inefficiency')).toBeInTheDocument();
      expect(screen.getByText('Client Frustration')).toBeInTheDocument();
      expect(screen.getByText('Work Interruption')).toBeInTheDocument();
    });

    it('displays correct descriptions for each problem area', () => {
      render(<ServicesHeroSection />);
      
      expect(screen.getByText(/partners and senior staff spending 25\+ hours weekly/i)).toBeInTheDocument();
      expect(screen.getByText(/high-value professionals doing low-value work/i)).toBeInTheDocument();
      expect(screen.getByText(/delayed responses and communication gaps/i)).toBeInTheDocument();
      expect(screen.getByText(/constant context switching disrupting deep work/i)).toBeInTheDocument();
    });

    it('has proper card structure with icons', () => {
      render(<ServicesHeroSection />);
      
      // Check that cards have the expected structure
      const problemCards = screen.getAllByRole('heading', { level: 3 });
      expect(problemCards).toHaveLength(4);
      
      // Each problem card should have an icon container (circular divs for icons)
      const iconContainers = document.querySelectorAll('.w-12.h-12.mx-auto.mb-4');
      expect(iconContainers).toHaveLength(4);
    });
  });

  describe('Responsive Design', () => {
    it('applies responsive grid classes', () => {
      render(<ServicesHeroSection />);
      
      const gridContainer = document.querySelector('.grid');
      expect(gridContainer).toHaveClass('grid-cols-1', 'md:grid-cols-2', 'xl:grid-cols-4');
    });

    it('applies responsive typography classes', () => {
      render(<ServicesHeroSection />);
      
      const mainHeading = screen.getByRole('heading', { 
        name: /intelligent workflow automation for professional services/i 
      });
      expect(mainHeading).toHaveClass('text-hero');
      
      const subheading = screen.getByText(/reclaim 150\+ hours per employee annually/i);
      expect(subheading).toHaveClass('text-subhero');
    });
  });

  describe('Styling and Theme Integration', () => {
    it('applies brand design system classes', () => {
      render(<ServicesHeroSection />);
      
      const mainHeading = screen.getByRole('heading', { 
        name: /intelligent workflow automation for professional services/i 
      });
      expect(mainHeading).toHaveClass('text-white');
      
      const problemHeading = screen.getByRole('heading', { 
        name: /the hidden cost of email-driven workflows/i 
      });
      expect(problemHeading).toHaveClass('text-brand-navy');
    });

    it('applies container and spacing utilities', () => {
      render(<ServicesHeroSection />);
      
      const containerElements = document.querySelectorAll('.container-brand');
      expect(containerElements.length).toBeGreaterThan(0);
      
      const sectionSpacing = document.querySelector('.section-spacing');
      expect(sectionSpacing).toBeInTheDocument();
    });
  });

  describe('Accessibility', () => {
    it('has proper heading hierarchy', () => {
      render(<ServicesHeroSection />);
      
      const h1 = screen.getByRole('heading', { level: 1 });
      const h2 = screen.getByRole('heading', { level: 2 });
      const h3Elements = screen.getAllByRole('heading', { level: 3 });
      
      expect(h1).toBeInTheDocument();
      expect(h2).toBeInTheDocument();
      expect(h3Elements).toHaveLength(4);
    });

    it('provides meaningful text content for screen readers', () => {
      render(<ServicesHeroSection />);
      
      // Check that all text content is meaningful and descriptive
      expect(screen.getByText(/intelligent workflow automation for professional services/i))
        .toBeInTheDocument();
      expect(screen.getByText(/professional services focus/i)).toBeInTheDocument();
      expect(screen.getByText(/the hidden cost of email-driven workflows/i)).toBeInTheDocument();
    });

    it('has proper semantic structure', () => {
      render(<ServicesHeroSection />);
      
      const sections = document.querySelectorAll('section');
      expect(sections).toHaveLength(2); // Hero section and problem statement section
    });
  });

  describe('Custom className prop', () => {
    it('applies custom className when provided', () => {
      render(<ServicesHeroSection className="custom-test-class" />);
      
      const component = document.querySelector('.custom-test-class');
      expect(component).toBeInTheDocument();
    });

    it('works without className prop', () => {
      render(<ServicesHeroSection />);
      
      // Should render without errors
      expect(screen.getByRole('heading', { 
        name: /intelligent workflow automation for professional services/i 
      })).toBeInTheDocument();
    });
  });

  describe('Integration with Design System', () => {
    it('uses Inter font for headings via font-heading class', () => {
      render(<ServicesHeroSection />);
      
      // Check specific headings that should have font-heading class
      const problemHeading = screen.getByRole('heading', { 
        name: /the hidden cost of email-driven workflows/i 
      });
      expect(problemHeading).toHaveClass('font-heading');
      
      const problemCardHeadings = screen.getAllByRole('heading', { level: 3 });
      problemCardHeadings.forEach(heading => {
        expect(heading).toHaveClass('font-heading');
      });
    });

    it('applies brand color variables correctly', () => {
      render(<ServicesHeroSection />);
      
      // Check the financial callout container has gradient background
      const financialCallout = screen.getByText(/\$236,000 to \$330,000/).closest('div');
      expect(financialCallout?.parentElement).toHaveClass('bg-gradient-accent');
    });
  });
});

describe('ServicesHeroSection Snapshot', () => {
  it('matches snapshot for visual regression testing', () => {
    const { container } = render(<ServicesHeroSection />);
    expect(container.firstChild).toMatchSnapshot();
  });
});