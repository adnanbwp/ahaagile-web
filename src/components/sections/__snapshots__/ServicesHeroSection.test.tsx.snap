// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesHeroSection Snapshot matches snapshot for visual regression testing 1`] = `
<div
  class=""
>
  <section
    class="relative py-20 lg:py-24"
    style="background: linear-gradient(135deg, var(--brand-navy) 0%, var(--brand-navy-light) 100%);"
  >
    <div
      class="absolute inset-0 opacity-10"
    >
      <div
        class="absolute inset-0 bg-gradient-to-r from-transparent via-white to-transparent transform -skew-y-1"
      />
    </div>
    <div
      class="container-brand relative z-10"
    >
      <div
        class="text-center max-w-4xl mx-auto"
      >
        <div
          class="mb-6"
        >
          <div
            class="inline-flex items-center rounded-full border transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 hover:bg-shadcn-secondary/80 bg-white/10 text-white border-white/20 text-sm font-medium px-4 py-2"
          >
            Professional Services Focus
          </div>
        </div>
        <h1
          class="text-hero mb-6 text-white"
        >
          Intelligent Workflow Automation for Professional Services
        </h1>
        <p
          class="text-subhero mb-8 text-gray-200 max-w-3xl mx-auto"
        >
          Reclaim 150+ Hours Per Employee Annually Through Intelligent Email Automation
        </p>
      </div>
    </div>
  </section>
  <section
    class="section-spacing bg-gradient-to-br from-secondary-50 to-white"
  >
    <div
      class="container-brand"
    >
      <div
        class="text-center max-w-4xl mx-auto mb-16"
      >
        <h2
          class="text-3xl lg:text-4xl font-heading font-bold text-brand-navy mb-6"
        >
          The Hidden Cost of Email-Driven Workflows
        </h2>
        <div
          class="bg-gradient-accent rounded-xl p-8 mb-8 text-center"
        >
          <div
            class="text-4xl lg:text-5xl font-bold text-brand-navy mb-2"
          >
            $236,000 to $330,000
          </div>
          <p
            class="text-lg font-medium text-brand-navy-dark"
          >
            Annual productivity loss for a typical 3-accountant professional services firm
          </p>
        </div>
        <p
          class="text-lg text-secondary-600 leading-relaxed"
        >
          Email-driven workflows are silently draining your firm's profitability. While your team juggles client communications, document management, and administrative tasks through scattered email threads, valuable billable hours disappear into operational inefficiency.
        </p>
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-4 gap-6"
      >
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:card-brand group transition-all duration-300"
        >
          <div
            class="p-6 text-center"
          >
            <div
              class="w-12 h-12 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center group-hover:bg-white/20 transition-colors duration-300"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-clock w-6 h-6 text-brand-navy group-hover:text-white"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 6v6l4 2"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
              </svg>
            </div>
            <h3
              class="text-lg font-heading font-semibold text-brand-navy group-hover:text-white mb-2 transition-colors duration-300"
            >
              Time Drain
            </h3>
            <p
              class="text-sm text-secondary-600 group-hover:text-gray-200 transition-colors duration-300"
            >
              Partners and senior staff spending 25+ hours weekly on email management and administrative tasks instead of billable client work.
            </p>
          </div>
        </div>
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:card-brand group transition-all duration-300"
        >
          <div
            class="p-6 text-center"
          >
            <div
              class="w-12 h-12 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center group-hover:bg-white/20 transition-colors duration-300"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-dollar-sign w-6 h-6 text-brand-navy group-hover:text-white"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="12"
                  x2="12"
                  y1="2"
                  y2="22"
                />
                <path
                  d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
                />
              </svg>
            </div>
            <h3
              class="text-lg font-heading font-semibold text-brand-navy group-hover:text-white mb-2 transition-colors duration-300"
            >
              Billing Inefficiency
            </h3>
            <p
              class="text-sm text-secondary-600 group-hover:text-gray-200 transition-colors duration-300"
            >
              High-value professionals doing low-value work, reducing billable hour capacity and directly impacting revenue potential.
            </p>
          </div>
        </div>
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:card-brand group transition-all duration-300"
        >
          <div
            class="p-6 text-center"
          >
            <div
              class="w-12 h-12 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center group-hover:bg-white/20 transition-colors duration-300"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-users w-6 h-6 text-brand-navy group-hover:text-white"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                />
                <path
                  d="M16 3.128a4 4 0 0 1 0 7.744"
                />
                <path
                  d="M22 21v-2a4 4 0 0 0-3-3.87"
                />
                <circle
                  cx="9"
                  cy="7"
                  r="4"
                />
              </svg>
            </div>
            <h3
              class="text-lg font-heading font-semibold text-brand-navy group-hover:text-white mb-2 transition-colors duration-300"
            >
              Client Frustration
            </h3>
            <p
              class="text-sm text-secondary-600 group-hover:text-gray-200 transition-colors duration-300"
            >
              Delayed responses and communication gaps leading to client dissatisfaction and potential account losses.
            </p>
          </div>
        </div>
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm card-elevated hover:card-brand group transition-all duration-300"
        >
          <div
            class="p-6 text-center"
          >
            <div
              class="w-12 h-12 mx-auto mb-4 bg-gradient-accent rounded-full flex items-center justify-center group-hover:bg-white/20 transition-colors duration-300"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-zap w-6 h-6 text-brand-navy group-hover:text-white"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
                />
              </svg>
            </div>
            <h3
              class="text-lg font-heading font-semibold text-brand-navy group-hover:text-white mb-2 transition-colors duration-300"
            >
              Work Interruption
            </h3>
            <p
              class="text-sm text-secondary-600 group-hover:text-gray-200 transition-colors duration-300"
            >
              Constant context switching disrupting deep work and reducing overall productivity and quality of deliverables.
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
